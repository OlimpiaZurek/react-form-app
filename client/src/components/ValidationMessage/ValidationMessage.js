import React from 'react';
import PropTypes from 'prop-types';
import styles from './ValidationMessage.module.scss';

const ValidationMessage = ({ status, text }) => (
  <div className={`${styles.Message} 
    ${status ? styles.Success : styles.Error}`}
  >
    {text}
  </div>
);

ValidationMessage.propTypes = {
  status: PropTypes.string,
  text: PropTypes.string,
};

export default ValidationMessage;

