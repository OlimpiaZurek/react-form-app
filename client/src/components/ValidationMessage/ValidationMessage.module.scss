@import '../../styles/global.scss';

.Message {
    font-size: 1.2em;
    text-align: center;
}

.Success {
    color: $main-color;
}

.Error {
    color: $danger-color;
}
