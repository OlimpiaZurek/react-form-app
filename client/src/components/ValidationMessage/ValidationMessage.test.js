import React from 'react';
import { shallow } from 'enzyme';
import ValidationMessage from './ValidationMessage';

const props = {
  status:'test',
  message: 'test'
};

describe('<ValidatonMessage', () => {
  it('should be defined', () => {
    expect(ValidationMessage).toBeDefined();
  });

  it('should render corectly', () => {
    const wrapper = shallow(
      <ValidationMessage {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  }); 
})
