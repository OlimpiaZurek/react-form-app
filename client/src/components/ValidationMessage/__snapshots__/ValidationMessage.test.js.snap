// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidatonMessage should render corectly 1`] = `ShallowWrapper {}`;
